File: typescript/agentkit/src/action-providers/cdp-v2/schemas.ts
import { z } from "zod";

/**
 * Input schema for request faucet funds action.
 */
export const RequestFaucetFundsV2Schema = z
  .object({
    assetId: z.string().optional().describe("The optional asset ID to request from faucet"),
  })
  .strip()
  .describe("Instructions for requesting faucet funds");

File: python/coinbase-agentkit/coinbase_agentkit/action_providers/cdp/schemas.py
"""Schemas for CDP action providers."""

from pydantic import BaseModel, Field


class RequestFaucetFundsSchema(BaseModel):
    """Input schema for requesting faucet funds."""

    asset_id: str | None = Field(
        None,
        description="The optional asset ID to request from faucet",
    )
